gobblin-core/src/main/java/gobblin/writer/PartitionedDataWriter.java

```diff
@@ -162,16 +172,20 @@ public long bytesWritten() throws IOException {
   }
 
   @Override
-  public void close() throws IOException {
+  public void close()
+      throws IOException {
     this.closer.close();
   }
 
-  private DataWriter<D> createPartitionWriter(GenericRecord partition) throws IOException {
+  private DataWriter<D> createPartitionWriter(GenericRecord partition)
+      throws IOException {
     if (!this.builder.isPresent()) {
       throw new IOException("Writer builder not found. This is an error in the code.");
     }
-    return this.builder.get().forPartition(partition).withWriterId(this.baseWriterId + "_" + this.writerIdSuffix++)
+    DataWriter dataWriter =  this.builder.get().forPartition(partition).withWriterId(this.baseWriterId + "_" + this.writerIdSuffix++)
         .build();
+    this.isSpeculativeAttemptAware = this.isDataWriterForPartitionSafe(dataWriter);
```

Comment:
You should "and" the safeness as there might be multiple writers instantiated for different partitions.